### ISSUE TITLE
Error: tried to put an item (arrow_fire_hardened_fletched, amount 1) in a container (recurbow) that cannot contain it

### ISSUE THREAD

**Issue by TheLightTwist** [CleverRaven/Cataclysm-DDA#53319]
### Describe the bug

Got following error when tried to place some arrows in inventory with 'g':

DEBUG    : tried to put an item (arrow_fire_hardened_fletched, amount 1) in a container (recurbow) that cannot contain it

 FUNCTION : int item::fill_with(const item&, int, bool, bool, bool)
 FILE     : src/item.cpp
 LINE     : 9000
 VERSION  : 0.F

### Steps to reproduce

1) Take some bow (but looking on text of that error, it must occur with crossbows too, I guess) in inventory
2) Try to pick more arrows than your quiver or the whole inventory can contain.

### Expected behavior

No error, obviously

### Versions and configuration

OS: Arch Linux x64
Game ver.: 0.F.2-1 tiles
Language: Russian

**Comment by Nymall** [comment]
Unable to reproduce - C117979

The game is handling reloading arrow_heavy weird - I can't actively reload the bow to that arrow, but I'm not getting any errors and  I can quick-fire if it is on the ground next to me, but I can't actively load the arrow. All other arrows work with no problems.

**Comment by hexagonrecursion** [comment]
Confirmed

1. [W]ear a modern recurve bow [id: recurbow]
2. [W]ear a quiver [id: quiver]
3. [g]rab 30 crude wooden arrows [id: arrow_fire_hardened_fletched]

 DEBUG    : tried to put an item (arrow_fire_hardened_fletched, amount 1) in a container (recurbow) that cannot contain it

 FUNCTION : int item::fill_with(const item&, int, bool, bool, bool)
 FILE     : src/item.cpp
 LINE     : 10373
 VERSION  : 9473cf6

- OS: Linux
    - OS Version: LSB Version: :core-4.1-amd64:core-4.1-noarch; Distributor ID: Fedora; Description: Fedora release 34 (Thirty Four); Release: 34; Codename: ThirtyFour; 
- Game Version: 9473cf6 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by LeahLuong** [comment]
This issue seems to apply to more than just archery. Items w/ charges & containers w/ insufficient storage seem to trigger this error. See #54445.

**Comment by psuedolet** [comment]
Reproduced with 7-round shotgun and compound hunting bow.
Only got the error when picking up a stack of ammo used by the equipped weapon while the character only had volume available for part of the stack.
If the ammo didn't match the equipped weapon, there was no weapon equipped, or there was sufficient inventory space, I did not get the error.
- OS: Windows
    - OS Version: 10.0.19044.1645 (21H2)
- Game Version: 0.F-7845-gd11f77a64b [64-bit]
